tools/crldps/main.go

```diff
@@ -0,0 +1,116 @@
+// crldps generates the list of CRL Distribution Point URIs for a given issuing
+// CA and number of shards. It fetches and validates all of those shards. If
+// it doesn't encounter any errors, it pretty-prints the list of all CRLDPs for
+// disclosure in CCADB.
+package main
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/crl/idp"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+
+	fs := flag.NewFlagSet("crldps", flag.ContinueOnError)
+	caPath := fs.String("ca", "", "path to an issuing intermediate CA certificate (required)")
+	numShards := fs.Int("shards", 128, "number of CRL shards issued by the CA")
+	err := fs.Parse(os.Args[1:])
+	if err != nil || len(fs.Args()) != 0 {
+		log.Println("Incorrect command line flags; usage:")
+		fs.PrintDefaults()
+		os.Exit(1)
+	}
+
+	issuer, err := core.LoadCert(*caPath)
+	if err != nil {
+		log.Fatalf("Failed to load issuer certificate from %q: %s", os.Args[1], err)
+	}
+
+	crldpBase := fmt.Sprintf("http://%s.c.lencr.org", strings.ToLower(issuer.Subject.CommonName))
+
+	var (
+		anyErr bool
+		crldps []string
+	)
+	for shard := range *numShards {
+		// We 1-index our CRL shards.
+		crldp := fmt.Sprintf("%s/%d.crl", crldpBase, shard+1)
+
+		resp, err := http.Get(crldp)
```

Comment:
According to [the docs](https://pkg.go.dev/net/http#Get):

> Get is a wrapper around DefaultClient.Get

This can be a problem because the `http.DefaultClient` doesn't have a timeout. If a shard is unresponsive, I believe the tool will just hang. Instead, you can construct a client with a timeout (outside of this loop) and use it:

```go
client := &http.Client{Timeout: 10 * time.Second}
```